datasets/swedish_medical_ner/swedish_medical_ner.py

```diff
@@ -0,0 +1,198 @@
+                            "start": datasets.Value("int32"),
+                            "end": datasets.Value("int32"),
+                            "text": datasets.Value("string"),
+                            "type": datasets.Value("string"),
+                        }
+                    ),
+                }
+            )
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage=_URL,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        urls_to_download = {
+            "wiki": _DATA_URL + "Wiki_annotated_60.txt",
+            "lt": _DATA_URL + "LT_annotated_60.txt",
+            "1177": _DATA_URL + "1177_annotated_sentences.txt",
+        }
+        downloaded_files = dl_manager.download_and_extract(urls_to_download)
+
+        if self.config.name == "wiki":
+            return [
+                datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"filepath": downloaded_files["wiki"]})
+            ]
+        elif self.config.name == "lt":
+            return [
+                datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"filepath": downloaded_files["lt"]})
+            ]
+        elif self.config.name == "1177":
+            return [
+                datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"filepath": downloaded_files["1177"]})
+            ]
+
+    def _generate_examples(self, filepath):
+        """Yields examples as (key, example) tuples."""
+
+        def find_type(s, e):
+            if (s == "(") and (e == ")"):
+                return "Disorder and Finding"
+            elif (s == "[") and (e == "]"):
+                return "Pharmaceutical Drug"
+            elif (s == "{") and (e == "}"):
+                return "Body Structure"
+            else:
+                return ""
+
+        pattern = r"\[([^\[\]()]+)\]|\(([^\[\]()]+)\)|\{([^\[\]()]+)\}"
+        with open(filepath, encoding="utf-8") as f:
+            for id_, row in enumerate(f):
+                sentence = row.replace("\n", "")
+
+                if self.config.name == "1177":
+                    targets = [
+                        {
+                            "start": m.start(0),
+                            "end": m.end(0),
+                            "text": sentence[m.start(0) + 2 : m.end(0) - 2],
+                            "type": find_type(sentence[m.start(0)], sentence[m.end(0) - 1]),
+                        }
+                        for m in re.finditer(pattern, sentence)
+                    ]
+                    yield id_, {
+                        "sid": self.config.name + "_" + str(id_),
+                        "sentence": sentence,
+                        "entities": targets if targets else [{"start": 999, "end": 999, "text": "", "type": ""}],
```

Comment:
Maybe we can just have an empty list here ?
```suggestion
                        "entities": targets if targets else [],
```

Reply:
Done.